src/network/networkd-dhcp6.c

```diff
@@ -42,25 +42,28 @@ bool link_dhcp6_pd_is_enabled(Link *link) {
         return link->network->dhcp6_pd;
 }
 
-static int dhcp6_pd_resolve_uplink(Link *link, Link **ret) {
+static bool dhcp6_pd_is_uplink(Link *link, Link *target, bool accept_auto) {
+        assert(link);
+        assert(target);
+
+        if (!link_dhcp6_pd_is_enabled(link))
+                return false;
+
```

Comment:
`assert(link->network)` or check

Reply:
`link_dhcp6_pd_is_enabled()` already checks that.